### ISSUE TITLE
databases-for-mysql user creation does not work

### ISSUE THREAD

**Issue by kulmannf** [IBM-Cloud/terraform-provider-ibm#4765]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform IBM Provider Version

```
terraform -v
Terraform v1.5.5
on darwin_arm64
+ provider registry.terraform.io/ibm-cloud/ibm v1.56.1
```

### Affected Resource(s)

* ibm_database

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
data "ibm_resource_group" "group" {
  name = "Default"
}

resource "ibm_database" "test" {
  name              = "test"
  plan              = "standard"
  location          = "eu-de"
  service           = "databases-for-mysql"
  resource_group_id = data.ibm_resource_group.group.id

  adminpassword = "password12"

  users {
    name     = "user123"
    password = "password12"
    type     = "database"
  }
}

```

### Debug Output

This is the trace output of trying to create a user on an already existing instance, but the initial creation also doesn't work: https://gist.github.com/kulmannf/067f87e884d11a8d9e06effb4c539657

### Expected Behavior

The databases for mysql instance is created and the `user123` exists and can be used

### Actual Behavior

The user does not exist (Output generated with admin user):
```
mysql> SELECT DISTINCT GRANTEE FROM information_schema.user_privileges;
+--------------------------------+
| GRANTEE                        |
+--------------------------------+
| 'ibm-backup'@'127.0.0.1'       |
| 'ibm'@'localhost'              |
| 'mysql.infoschema'@'localhost' |
| 'mysql.session'@'localhost'    |
| 'mysql.sys'@'localhost'        |
| 'admin'@'%'                    |
| 'ibm-replication'@'%'          |
| 'ibm-monitor'@'%'              |
+--------------------------------+
8 rows in set (0.01 sec)
```
Trying to authenticate with the credentials for the user results in a Access Denied error message.

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by kavya498** [comment]
Looks like the code is not entering usercreate block at all [here](https://github.com/IBM-Cloud/terraform-provider-ibm/blob/master/ibm/service/database/resource_ibm_database.go#L3155)
When debug logs is enable we only see PATCH call which tries to update user password. There is call to POST which creates user

enable TF_LOG=debug and reproduce bug using above TF snippet.

Even the user update doesn't work
Try to update user1 -->user2

Plan says it ll delete user1 and will create user2. --> which is expected. 
But below is what apply does:
It did call DELETE  and there is a PATCH call after that and no POST to create users.

Although I was wondering what DELETE did as tf said it created user1 in previous apply but there is no user1 user in Database.

I can share the debug logs on slack if you want to have a look.

@obai-1 @omaraibrahim  FYI

**Comment by shipra101** [comment]
Hi @omaraibrahim 
We see [this](https://github.ibm.com/compose/compose/issues/40412) bug is closed. Can you confirm if the issue is resolved? 
Thanks in advance!

**Comment by omaraibrahim** [comment]
@shipra101 Yes the bug has been fixed, but the fix has not been release yet unfortunately 😞 I'll let you know when the patch is live.

**Comment by Xxiao-ma** [comment]
Hi @omaraibrahim, do you have any news about the release? We're currently blocked by this since 3 weeks

**Comment by omaraibrahim** [comment]
@Xxiao-ma Unfortunately there is no news 😞 There is a workaround where you can use the CLI as mentioned here: https://github.com/IBM-Cloud/terraform-provider-ibm/issues/4765#issuecomment-1697426469

If that doesn't work please lmk.

**Comment by niranjan-ibm** [comment]
Hi @omaraibrahim, any updates on this yet? We already have a workaround for this, but we now need a permanent fix which would be possible whenever this bug fix is available in your release.
